util/settings/settings.go

```diff
@@ -205,8 +205,36 @@ const (
 	anonymousUserEnabledKey = "users.anonymous.enabled"
 )
 
-// SettingsManager holds config info for a new manager with which to access Kubernetes ConfigMaps.
-type SettingsManager struct {
+type SettingsManager interface {
```

Comment:
I agree that it is painful to test code which depends on SettingManager. 

I've faced that same issue recently and realized that no Argo CD part uses all methods of settings manager. Instead of introducing one giant interface I would suggest to keep setting manager as is and replace reference to it with interfaces that contains subset of required methods.

E.g. `argo.db` uses only `GetRepositories`, `SaveRepositories` and `GetSecretsLister` . So we can define interface e.g. `SettingsStore` that has three methods and replace the reference to  SettingsManager pointer with reference to `SettingsStore`. At the same time we can generate `SettingsStore` mock and simplify `argo.db` tests

Reply:
good idea - v2?